File: 프로그래머스/lv2/131127. 할인 행사/할인 행사.py
from collections import defaultdict
def solution(want, number, discount):
    answer = 0
    goods = dict()
    discount_goods  = defaultdict(int)
    cnt = 0
    for i in range(len(want)) :
        goods[want[i]] = number[i]
    for i in range(len(discount)-9) :
        discount_goods = defaultdict(int)
        cnt += 1
        for j in range(i,i+10) :
            discount_goods[discount[j]] += 1
        flag = True
        for k in range(len(want)) :
            if discount_goods[want[k]] != number[k] :
                flag = False
                break
        if flag == True :
            answer += 1

    return answer

File: 프로그래머스/2/131127. 할인 행사/할인 행사.java
import java.util.*;

class Solution {
    public int solution(String[] want, int[] number, String[] discount) {
        int answer = 0;
        HashMap<String,Integer> map = new HashMap<>();
        for (int i=0; i < want.length; i++) {
            map.put(want[i], number[i]);
        }
        
        for (int i=0; i < discount.length-9; i++) {
            HashMap<String,Integer> map2 = new HashMap<>();
            boolean flag = true;
            for (int j=0; j < 10; j++) {
                map2.put(discount[i+j], map2.getOrDefault(discount[i+j],0) + 1);            
            }
            // System.out.println(map.get(discount[i]));
            for (String k : map2.keySet()) {
                if (map2.get(k) != map.get(k)) {
                    flag = false;
                    break;
                }
            }
            
            if (flag) {
                answer++;
            }
        }
        return answer;
    }
}
